src/shared/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import Icon from '../Icon'
+import { PaginationWrapper, PaginationButton } from './Pagination.style'
+
+type PaginationProps = {
+  itemsPerPage?: number
+  totalCount?: number
+  currentPage?: number
+  maxPaginationLinks?: number
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
```

Comment:
Let's make this `onChange` / `onPageChange` and pass the new page number to the consumer